Zend/zend_ini.c

```diff
@@ -634,6 +662,18 @@ static zend_ulong zend_ini_parse_quantity_internal(zend_string *value, zend_ini_
 			smart_str_append_escaped(&invalid, ZSTR_VAL(value), ZSTR_LEN(value));
 			smart_str_0(&invalid);
 
+			*errstr = zend_strpprintf(0, "Invalid quantity \"%s\": no digits after base prefix, interpreting as \"0\" for backwards compatibility",
+							ZSTR_VAL(invalid.s));
+
+			smart_str_free(&invalid);
+			return 0;
+		}
+		if (UNEXPECTED(digits != zend_ini_consume_quantity_prefix(digits, str_end))) {
```

Comment:
seems to me could be interpreted/simplified slightly as :

`if (UNEXPECTED(digits == str_end) || UNEXPECTED(digits != zend_ini_consume_quantity_prefix(digits, str_end)) {
     goto invalid_quantity;
}`

eventually even having a `const char *errmsg` so line 614 can be address too.

unless I misread. What do you think ?

Reply:
The condition change makes sense, not sure I understand what you mean about line 614?

Reply:
Line like 614 is just displaying the wrong case in the same fashion, only the error message changes compared to 665.

Reply:
Ah right :+1: